### DISCUSSION TITLE
F' on multi-core systems

### DISCUSSION THREAD

**Discussion by DpinkyandDbrain** [nasa/fprime#1809] [Q&A] [answered]
Does anyone have experience using F' on a multi-core system running as SMP? And, if there are any lessons learned? I expect that F' does support multi-core and SMP as long as you use it correctly. I am curious if there are any components though that just won't work safely on multicore without a lot of mods.

**Comment by LeStarch** [comment, marked as answer]
The F´ architecture is generally good about multiple cores. I know we have run on multi core systems before.   `async` components use queues to pass messages between threads. This architecture is fairly good at allowing concurrency.  The best advice I can give here is to embrace the port-call and `async` components and not "work around" them as then you'd be solving these problems again.
